text/0000-my-feature.md

```diff
@@ -0,0 +1,123 @@
+- Enable developers to quickly identify which buildpacks are applicable to their application without waiting for the entire build process to complete, or having to `Ctrl+C` midway through.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This provides a high level overview of the feature.
+
+- Define any new terminology.
+- Define the target persona: buildpack author, buildpack user, platform operator, platform implementor, and/or project contributor.
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Ideally, the user should run something like `pack detect --path ./path/to/project --builder builder:name` and it should run the analyze binary, followed by the detect binary in the lifecycle and output the logs / output of it. Preferrably, can also output the `group.toml` to a directory specified with `--detect-output-dir`. The reason to run the analyze binary is to get information about the run image that may impact the outcome of detect via CNB_TARGET_* environment variables.
+
+The following flags should be supported and they will work more or less like `pack build`.
+
+|Short|Long|type|
+|----|-------------|----|
+|-B |--builder|string|
+|-b |--buildpack|strings|
+|-r |--buildpack-registry|string|
+||--detect-output-dir|string|
+|-d|--descriptor|string|
+||--docker-host|string|
+|-e|--env|stringArray|
+||--env-file|stringArray|
+||--extension|strings|
+||--gid|int|
+|-h|--help|
+||--lifecycle-image|string|
+||--network|string|
+|-p|--path|string|
+||--post-buildpack|stringArray|
+||--pre-buildpack|stringArray|
+||--pull-policy|string|
+||--run-image|string|
+||--uid|int|
+||--workspace|string|
+
+
+# Migration
+[migration]: #migration
+
+This feature does not introduce any breaks to public APIs or compatibility. It provides additional functionality within the existing Cloud Native Buildpacks CLI tooling, enhancing the developer experience without requiring changes to existing workflows or configurations.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Alternatives
+[alternatives]: #alternatives
+
+Initially thought of implementing this through something like `pack build --detect`. But after further discussion with @jjbustamante and for the following reasons, decided to do this functionality to a new command.
+- The main use case of `pack build` is to create OCI images, and detect is just a binary in the lifecycle, so it doesn't make much sense to include it in there.
+- To avoid making the mostly used `pack build` command overly complicated.
+
+Also, instead of `pack detect`, something like `pack execute --phase detect` can also be done, where the user can select exclusively what phase they need to run. Can start by implementing just the `detect` phase.
```

Comment:
This might be worth doing. It could be nice for integration testing purposes to be able to run just the build or just the generate phases of a buildpack. Not that we'd include those into this RFC, but this would leave the CLI open for that possibility.

When it comes to integration testing buildpacks with complicated build plans, right now, you have three choices:

1. Don't integration test them.
2. Supply all the buildpacks required to run the buildpack under test. That way all the set up and requirements for this buildpack are complete.

If we could run build or generate specifically, then that would open up the ability to mock out the requirements and run more focused, faster integration tests.

Anyway, that is all just to say I would be in support of this CLI invocation as it leaves open the possibility of running other phases directly.

Reply:
Yes, it looks like the better path to take